### DISCUSSION TITLE
Hot to set .navigationDestination in second view

### DISCUSSION THREAD

**Discussion by hellotimw** [pointfreeco/swift-composable-architecture#2917] [Q&A] [answered]
I'm having trouble setting .navigationDestination in the second view of my SwiftUI app. I'm using TCA and the structure of my View is:  FeedListView -> ItemListView -> DetailView

I want to use `.navigationDestination` in the `List()` of `ItemListView` to navigate to `DetailView`. I've already used `NavigationStack` and `.forEach(\.path, action:\.path)` in `FeedListView` to navigate. I'm not sure how to write the corresponding code in `ItemListFeature`.

I tried the following code, but it gives me an error: **A NavigationLink is presenting a value of type “Component” but there is no matching navigationDestination declaration visible from the location of the link. The link cannot be activated.**

```swift
// ItemListFeature
struct State: Equatable {
    @Presents var itemDetail: ItemDetailFeature.State?
}
enum Action {
    case itemDetail(PresentationAction<ItemDetailFeature.Action>)
}

var body: some ReduceOf<Self> {
    Reduce { state, action, environment in
        ...
    }
    .ifLet(\.$itemDetail, action: \.itemDetail) {
        ItemDetailFeature()
    }
}

// ItemListView
List(store.items) { item in
    NavigationLink(state: ItemDetailFeature.State(item: item)) {
        itemCellView(item: item)
    }
}
.navigationDestination(item: $store.scope(state: \.itemDetail, action: \.itemDetail)) { store in
    ItemDetailView(store: store)
}
```
Could anyone please help me with this? I'm new to TCA and I'm trying to learn how to use it in my projects.

Additional information:
-  I'm using Xcode 15.0
- I'm using the 1.9.2 Composable Architecture library.

I would appreciate it if you could provide a solution that is clear and easy to understand. Thank you!

Best regards

**Comment by mbrandonw** [comment]
Hi @IDTIMW, you cannot use `NavigationLink(state:)` with `navigationDestination(item:)`. This is true of vanilla SwiftUI, it's not an issue with TCA. `NavigationLink(state:)` (and `NavigationLink(value:)` for that matter) only work with the version of `NavigationStack` that takes a path binding.

If you want to use `navigationDestination(item:)` you need to use a plain `Button` and send an action that populates the `itemDetail` state.

**Reply by hellotimw** [reply]
Thank you for your reply. @mbrandonw 

 I  used `NavigationStack` is in `FeedListView`, and I also used the `path` `@Bindable` store method in `FeedListView` to navigate to `ItemListView` according to the TCA tutorial.

In this case, how should I handle the navigation if I want to continue navigating from the `List()` in `ItemListView`? What I really want to know is, in TCA, what is the most recommended way to handle navigation for views with multiple levels (not sheets), such as continuing to navigate to a new page from `ItemDetailView`? Is it like you mentioned above, using the `Button` and send an action that populates the itemDetail state.?

Could you please provide some key code examples or tutorials for me to learn? I really appreciate your help.

**Reply by hellotimw** [reply]
Hi @mbrandonw 
I found the code in this [NavigateAndLoad.swift](https://github.com/pointfreeco/swift-composable-architecture/blob/main/Examples/CaseStudies/SwiftUICaseStudies/03-Navigation-Lists-NavigateAndLoad.swift#L82) which seems to be a reference for solving my problem. Use `NavigationLink` to handle my needs.

I need to try it out.

**Reply by mbrandonw** [reply, marked as answer]
You are mixing two styles of navigation. It seems you have a `NavigationStack` with a path somewhere in your app, which means you are using a collection of states to control what screens are being presented. And then you are also using an optional state to drive presenting the `ItemDetailFeature`. That is the problem you are having, and it would be the same problem if you were to remake this in vanilla SwiftUI.

If you want to push the `ItemDetailFeature` onto the stack, then the `ItemDetailFeature` needs to be integrated into your `Path` reducer, and then you can use `NavigationLink(state:)` to push the feature onto the stack.

Be sure to read our [navigation docs](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/navigation) to learn more, especially the [stack-based navigation docs](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/stackbasednavigation). There are also a few tutorials in the docs that show of stack-based navigation.

**Reply by hellotimw** [reply]
Thank you very much for your guidance @mbrandonw . After reading the articles you provided, I successfully solved my problem.

**Reply by michaelkariv** [reply]
@IDTIMW , can you please explain what the solution is? Your pattern seems to be my case as well
